packages/pwa-kit-react-sdk/src/ssr/universal/hooks.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useContext} from 'react'
+import {IsPrePassContext, ServerContext} from './contexts'
+
+/**
+ * Server context
+ * @typedef {Object} ServerContext
+ * @property {Object} req - Request object
+ * @property {Object} res - Response object
+ * @property {boolean} isServerSide
+ */
+
+/**
+ * Get the server context
+ * @returns {ServerContext} ServerContext object
+ *
+ * @example
+ * const {res, isServerSide} = useServerContext()
+ * if (isServerSide && query.error) { res.status(404) }
+ */
+export const useServerContext = () => {
+    const serverContext = useContext(ServerContext)
+    const isPrePass = useIsPrePass()
+
+    return {
+        ...serverContext,
+        isServerSide: Boolean(serverContext.req) && !isPrePass
```

Comment:
I understand what you are doing here. But I think this condition is better suited to your old callback implementation. For example, I read this as "isServerSide is true, if I'm on the server, but I'm not prepassing on the server". If this same condition was used with the callback method, then it would read something like "The callback is called only on the server but not during prepass".. It sounds a little easier to understand.